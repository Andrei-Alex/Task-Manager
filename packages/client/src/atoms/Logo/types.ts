import { IAppData } from "@/services/appData";

export type ILogo = {
  logo: {
    name?: string;
    src?: string;
  };
  appData?: IAppData;
};
