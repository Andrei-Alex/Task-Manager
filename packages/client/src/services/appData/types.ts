export interface IAppData {
  server: Partial<AppDataServer>;
  clientVersion: string | undefined;
}

export type AppDataServer = {
  serverStatus: number;
  dbName: string;
  dbStatus: string;
  serverVersion: string;
};
