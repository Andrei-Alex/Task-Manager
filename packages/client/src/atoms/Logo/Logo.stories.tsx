import type { Meta, StoryObj } from "@storybook/react";
import React from "react";
import { Logo } from "./Logo";

const meta: Meta<typeof Logo> = {
  component: Logo,
  title: "Atoms/Logo",
  tags: ["autodocs"],
  argTypes: {},
};
export default meta;
type Story = StoryObj<typeof Logo>;

export const Default: Story = {
  args: {
    logo: { name: "Kanban" },
    appData: {
      clientVersion: "1.6.8",
      server: { dbName: "dev", serverVersion: "1.9.6" },
    },
  },
};
