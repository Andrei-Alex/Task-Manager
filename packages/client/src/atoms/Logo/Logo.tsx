import React from "react";

const Logo = ({ logo }) => {
  return (
    <div>{logo.src ? <image href={logo.src} /> : <h1>{logo.name}</h1>}</div>
  );
};

export default Logo;
