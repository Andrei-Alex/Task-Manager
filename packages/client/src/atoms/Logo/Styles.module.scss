.logo {
  display: inline-flex;
  flex-direction: column;
  justify-content: space-around;
  font-family: sans-serif;
  padding: 1rem;

  h1 {
    margin: 0;
    width: fit-content;
    font-size: 4rem;
  }

  .dev, .data {
    display: flex;
    justify-content: space-between;
    font-size: 0.8rem;
  }
  .dev > p:first-of-type {
    color: red;
  }
  p:not(:last-child) {
    margin-right: 0.5rem;
  }

}