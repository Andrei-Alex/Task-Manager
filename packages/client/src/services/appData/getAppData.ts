import { AxiosError } from "axios";
import { appDataServerSideInstance, appDateInstance } from "@/constants";
import { AppDataServer } from "@/services/appData/types";

/**
 * This function fetches Back-end data using the appDataServerSideInstance.
 * @returns {Promise<Object|AxiosError>} A Promise that resolves to the fetched data or an AxiosError in case of an error.
 */
export const getAppData = async (): Promise<AppDataServer | AxiosError> => {
  try {
    const response = await appDateInstance({});
    return response.data;
  } catch (error) {
    console.error(error);
    return error as AxiosError;
  }
};

/**
 * This function fetches Next server-side data using the appDataServerSideInstance.
 * @returns {Promise<Object|AxiosError>} A Promise that resolves to the fetched data or an AxiosError in case of an error.
 */
export const getServerSideAppData = async () => {
  try {
    const response = await appDataServerSideInstance({});
    return response.data;
  } catch (error) {
    console.error(error);
    return error as AxiosError;
  }
};
