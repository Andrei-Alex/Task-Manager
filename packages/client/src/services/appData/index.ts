export * from "./types";
export { getAppData, getServerSideAppData } from "./getAppData";
