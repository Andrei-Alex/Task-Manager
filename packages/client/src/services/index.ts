export * from "./health";
export * from "./authentication/";
