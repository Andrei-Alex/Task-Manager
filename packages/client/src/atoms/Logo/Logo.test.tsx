import renderer from "react-test-renderer";
import React from "react";
import { IAppData } from "@/services/appData";
import { Logo as LogoComponent } from "./Logo";
import { Logo } from "../../../../libs/sharedTypes";
import { render } from "@testing-library/react";
import { screen } from "@testing-library/dom";

describe("Logo", () => {
  const logo: Logo = {
    name: "Kanban",
  };
  const appData: IAppData = {
    server: {
      serverVersion: "1.1.1",
      dbName: "dev",
    },
    clientVersion: "1.2.2",
  };

  it("renders correctly", () => {
    const tree = renderer
      .create(<LogoComponent logo={logo} appData={appData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render dev info", async () => {
    render(<LogoComponent logo={logo} appData={appData} />);
    expect(screen.getByText("Dev")).toBeInTheDocument();
    expect(
      screen.getByText(`Server:${appData.server.serverVersion}`)
    ).toBeInTheDocument();
    expect(
      screen.getByText(`Client:${appData.clientVersion}`)
    ).toBeInTheDocument();
  });
  it("should not render dev info", async () => {
    render(<LogoComponent logo={logo} />);
    const dev = screen.queryByText("Dev");
    const serverVersion = screen.queryByText(
      `Server:${appData.server.serverVersion}`
    );
    const clientVersion = screen.queryByText(`Client:${appData.clientVersion}`);
    expect(dev).toBeNull();
    expect(serverVersion).toBeNull();
    expect(clientVersion).toBeNull();
  });
});
