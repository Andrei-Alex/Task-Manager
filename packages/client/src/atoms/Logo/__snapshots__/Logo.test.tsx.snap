// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
<div
  className="logo"
>
  <h1>
    Kanban
  </h1>
  <div
    className="dev"
  >
    <p>
      Dev
    </p>
    <div
      className="data"
    >
      <p>
        Client: 
        1.2.2
      </p>
      <p>
        Server: 
        1.1.1
      </p>
    </div>
  </div>
</div>
`;
